cypress/integration/pages/mediaAssetPage/testsForCanonicalOnly.js

```diff
@@ -19,41 +19,51 @@ export const testsThatFollowSmokeTestConfigForCanonicalOnly = ({
     it('should render a media player', () => {
       cy.request(`${config[service].pageTypes[pageType].path}.json`).then(
         ({ body }) => {
-          const { assetUri } = body.metadata.locators;
-          const mediaBlock = body.content.blocks[0];
-          const isLiveStream = mediaBlock.type === 'version';
-          const serviceId = isLiveStream
-            ? mediaBlock.externalId
-            : mediaBlock.versions[0].versionId;
-          const language = appConfig[config[service].name][variant].lang;
-
-          cy.get(
-            `iframe[src*="${envConfig.avEmbedBaseUrl}/ws/av-embeds/cps${assetUri}/${serviceId}/${language}"]`,
-          ).should('be.visible');
+          const assetUriString = body.metadata.locators.assetUri;
+          if (assetUriString.split('/').length > 4) {
+            cy.log('Test skipped because legacy MAP');
+          } else {
+            const mediaBlock = body.content.blocks[0];
+            const isLiveStream = mediaBlock.type === 'version';
+            const serviceId = isLiveStream
+              ? mediaBlock.externalId
+              : mediaBlock.versions[0].versionId;
+            const language = appConfig[config[service].name][variant].lang;
+            const { assetUri } = body.metadata.locators;
+            cy.get(
+              `iframe[src*="${envConfig.avEmbedBaseUrl}/ws/av-embeds/cps${assetUri}/${serviceId}/${language}"]`,
+            ).should('be.visible');
+          }
         },
       );
     });
     it('should render an SMP player that is ready to play media', () => {
       cy.window().then(win => {
-        const media =
-          getBlockData('video', win.SIMORGH_DATA.pageData) ||
-          getBlockData('version', win.SIMORGH_DATA.pageData);
+        const assetUriString =
+          win.SIMORGH_DATA.pageData.metadata.locators.assetUri;
+        if (assetUriString.split('/').length > 4) {
+          cy.log('Test skipped because legacy MAP');
+        } else {
+          const media =
+            getBlockData('video', win.SIMORGH_DATA.pageData) ||
+            getBlockData('version', win.SIMORGH_DATA.pageData);
 
-        if (!media) throw new Error('no media');
+          if (!media) throw new Error('no media');
 
-        if (media) {
-          cy.get(
-            'div[class^="StyledVideoContainer"] iframe[class^="StyledIframe"]',
-          ).then($iframe => {
-            cy.wrap($iframe.prop('contentWindow'), {
-              // `timeout` only applies to the methods chained below.
-              // `its()` benefits from this, and will wait up to 8s
-              // for the mediaPlayer instance to become available.
-              timeout: 8000,
-            })
-              .its('embeddedMedia.playerInstances.mediaPlayer.ready')
-              .should('eq', true);
-          });
+          if (media) {
```

Comment:
This line (and line 66) aren't entirely needed - if we get this far, we know there's media, because we got past the check on line 51.

Functions stop running as soon as they hit a `return` statement or throw an `Error` 😄